Toolset/palettes/script editor/behaviors/revsecommoneditorbehavior.livecodescript

```diff
@@ -3109,3 +3214,1204 @@ private command __LookForPair pChar, pPairChar, pStep, @xCharNum
+         put pHandler && "of ${object:expression.chunk.object}" into tCompletion
+         put true into tMatchToken
+         break
+      case "function"
+         put "expression.function.lcs" into tClass
+         put pHandler & "(" into tCompletion
+         put "(" after tDisplayName
+         __ParseParams pHandlerInfo["parameters"], tDisplayName, tCompletion
+         put ")" after tDisplayName
+         put ")" after tCompletion
+         put true into tMatchToken
+         break
+      default
+         put "statement.command.lcs" into tClass
+         put pHandler into tCompletion
+         __ParseParams pHandlerInfo["parameters"], tDisplayName, tCompletion
+         put false into tMatchToken
+         break
+   end switch
+   
+   put the number of elements of xProviders[pPriority][tClass] + 1 into tIndex
+   put tDisplayName into xProviders[pPriority][tClass][tIndex]["displayname"]
+   put tCompletion into xProviders[pPriority][tClass][tIndex]["completion"]
+   put tClass into xProviders[pPriority][tClass][tIndex]["symbolclass"]
+   put tMatchToken into xProviders[pPriority][tClass][tIndex]["matchtoken"]
+   put "handler" into xProviders[pPriority][tClass][tIndex]["scope"]
+end __AddHandlerProvider
+
+/*
+
+Summary: Cancel the current provider search 
+
+*/
+
+private command __AutoCompleteCancelFind
+   if sLastFindMessageID is not empty then
+      cancel sLastFindMessageID
+      put empty into sLastFindMessageID
+   end if
+end __AutoCompleteCancelFind
+
+private command __CurrentTokens @rToken, @rLineTokens, @rChar, @rLine, @rLineChar
+   put word 4 of the selectedChunk into rChar
+   put word 2 of the selectedLine into rLine
+   local tLine
+   put the last line of (char 1 to rChar of field "Script" me) into tLine
+   if the last char of tLine is not space then
+      put the last word of tLine into rToken
+   end if
+   put word 1 to -1 of tLine into rLineTokens
+   put the number of chars of tLine into rLineChar
+end __CurrentTokens
+
+private command __CancelFindProviders
+   if there is a field "completions" of me then
+      hide field "completions" of me
+   end if
+   __AutoCompleteCancelFind
+   delete variable sFoundProviders
+end __CancelFindProviders
+
+private command __UpdateAutoCompleteList pUUID
+   local tChar, tLineChar, tLine, tToken, tLineTokens
+   __CurrentTokens tToken, tLineTokens, tChar, tLine, tLineChar
+   
+   if tToken is not empty then
+      local tSymbolClasses
+      put sPlaceholders[pUUID]["classes"] into tSymbolClasses
+      __AutoCompleteFindProviders sObjectID, tToken, tLineTokens, tLine,tSymbolClasses,the long id of me 
+   else
+      __CancelFindProviders
+   end if
+end __UpdateAutoCompleteList
+
+private function __IconForSymbolClass pSymbolClass
+   switch
+      case pSymbolClass contains "command"
+         return "<b>C</b>"
+      case pSymbolClass contains "function"
+         return "<b>F</b>"
+      case pSymbolClass contains "command"
+         return "<b>V</b>"
```

Comment:
Thanks @BerndN nice catch. Just a copy, paste error. BTW I'm looking for ideas on better chars for comment. I originally had `#` but then it looked odd in the list having `# -- comment`. So I went with something different `~`... But I'm open to ides.